reps/2022-03-10-plasma_alternative.md

```diff
@@ -0,0 +1,216 @@
+
+[**[Vineyard]**](https://github.com/v6d-io/v6d) is also an in-memory immutable data manager that provides out-of-the-box high-level abstraction and zero-copy in-memory sharing for distributed data in big data tasks. It aims to optimize the end-to-end performance of complex data analytics jobs on cloud platforms. It manages various data structures produced in the job, and cross-system data sharing can be conducted without serialization/deserialization overheads by decoupling metadata and data payloads.
+
+**Opportunities**: Vineyard can enhance the ray object store with more functionalities for modern data-intensive applications. To be specific, the benefits are four-fold:
+
+- O1: Sharing data with systems out of Ray. Vineyard is a daemon in cluster that any system can put/get objects into/from it.
+- O2: Composability. Objects will not be sealed twice when we create a new object form a existing one(e.g. add a new column for dataframe).
+- O3: Zero copy in put/get. Objects can be directly created in object store without copying or serialization.
+- O4: Reusable Routines (IO/Chunking/Partitions).
+
+**At the first step, we only focus on the O1**. Specifically speaking, Plasma store is currently an internal component of raylet, it shares the same fate with raylet. For systems that are not integrated with ray, they can not share objects via Plasma, thus have to share them by filesystem which will inevitably introduce heavy overheads if their data are complex (e.g. serializing a graph cost a long time.). While vineyard serves as a daemon in kubernete cluster.
+
+#### Key requirements:
+- Provide the options for users to select a third-party object store at `ray.init()`.
+- [O1] Provide a  Plasma-like client which forward the request to vineyard client. (vineyard server provide very similar functionality as plasma).
+
+### Should this change be within `ray` or outside?
+main `ray` project. Changes are made to Ray Core level.
+
+## Stewardship
+### Required Reviewers
+The proposal will be open to the public, but please suggest a few experience Ray contributors in this technical domain whose comments will help this proposal. Ideally, the list should include Ray committers.
+
+@ericl, @rkooo567, @scv119, @jjyao, @MissiontoMars
+
+### Shepherd of the Proposal (should be a senior committer)
+To make the review process more productive, the owner of each proposal should identify a **shepherd** (should be a senior Ray committer). The shepherd is responsible for working with the owner and making sure the proposal is in good shape (with necessary information) before marking it as ready for broader review.
+
+TBF
```

Comment:
remove this?

Reply:
Now the shepherds are @scv119 and @rkooo567, according to https://github.com/ray-project/ray/pull/22948